main/src/library/Map.flix

```diff
@@ -250,13 +266,84 @@ namespace Map {
     def foldRight[k, v, b](f: (v, b) -> b, s: b, m: Map[k, v]): b = foldRightWithKey((k, v, b) -> f(v, b), s, m)
 
     ///
-    /// Applies `f` to a start value `s` and all values in `m` going from right to left.
-    /// That is, the result is of the form: `f(v1, ...f(vn-1, f(vn, s)))`.
+    /// Applies `f` to a start value `s` and all key-value pairs in `m` going from right to left.
+    /// That is, the result is of the form: `f(k1, v1, ...f(kn-1, vn-1, f(kn, vn, s)))`.
     ///
     def foldRightWithKey[k, v, b](f: (k, v, b) -> b, s: b, m: Map[k, v]): b =
         let Map(xs) = m;
             List.foldRight((kv, y) -> f(fst(kv), snd(kv), y), s, xs)
 
+    ///
+    /// Alias for `reduceLeft`.
+    ///
+    def reduce[k, v](f: (v, v) -> v, m: Map[k, v]): Option[v] = reduceLeft(f, m)
+
+    ///
+    /// Alias for `reduceLeftWithKey`.
+    ///
+    def reduceWithKey[k, v](f: (k, v, k, v) -> (k, v), m: Map[k, v]): Option[(k, v)] = reduceLeftWithKey(f, m)
+
+    ///
+    /// Applies `f` to all values in `m` going from left to right until a single value `v` is obtained.  Returns `Some(v)`.
+    /// That is, the result is of the form: `Some(f(...f(f(v1, v2), v3)..., vn))`
+    /// Returns `None` if `m` is the empty map.
+    ///
+    def reduceLeft[k, v](f: (v, v) -> v, m: Map[k, v]): Option[v] = List.reduceLeftOpt(f, m.valuesOf())
```

Comment:
I thought we renamed `List.reduceLeftOpt` to `List.reduceLeft`?  but maybe not in this PR?